### ISSUE TITLE
Types are broken

### ISSUE THREAD

**Issue by mohd-akram** [valeriangalliat/fetch-cookie#84]
To reproduce:

`package.json`:

```json
{
  "type": "module",
  "dependencies": {
    "fetch-cookie": "^3.0.0",
    "typescript": "^5.3.3"
  }
}
```

`main.ts`:

```typescript
import fetchCookie from 'fetch-cookie';
export default fetchCookie;
```

Run `node_modules/.bin/tsc -m node16 --noEmit --strict main.ts`

Output:

```typescript
main.ts:1:25 - error TS7016: Could not find a declaration file for module 'fetch-cookie'. '/private/tmp/fetch-cookie-test/node_modules/fetch-cookie/esm/index.js' implicitly has an 'any' type.
  There are types at '/private/tmp/fetch-cookie-test/node_modules/fetch-cookie/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'fetch-cookie' library may need to update its package.json or typings.

1 import fetchCookie from 'fetch-cookie';
                          ~~~~~~~~~~~~~~


Found 1 error in main.ts:1
```

**Comment by valeriangalliat** [comment]
Thanks for reporting! This should be fixed with 3.0.1, let me know :)

**Comment by mohd-akram** [comment]
Thanks, works now!
